cogeo_mosaic/backends/base.py

```diff
@@ -64,3 +69,83 @@ def write(self):
     @abc.abstractmethod
     def update(self):
         """Update existing MosaicJSON on backend."""
+
+    def _update_quadkey(self, quadkey: str, dataset: List[str]):
+        """Update quadkey list."""
+        self.mosaic_def.tiles[quadkey] = dataset
+
+    def _update_metadata(self, bounds: List[float], version: str):
+        """Update bounds and center."""
+        self.mosaic_def.version = version
+        self.mosaic_def.bounds = bounds
+        self.mosaic_def.center = (
+            (bounds[0] + bounds[2]) / 2,
+            (bounds[1] + bounds[3]) / 2,
+            self.mosaic_def.minzoom,
+        )
+
+    def _update(
+        self,
+        features: Sequence[Dict],
+        accessor: Callable,
+        on_top: bool = True,
+        **kwargs
+    ):
+        """Update existing MosaicJSON on backend."""
+        version = self.mosaic_def.version
+        if version:
+            uversion = list(map(int, version.split(".")))
+            uversion[-1] += 1
+            version = ".".join(map(str, uversion))
+        else:
+            version = "1.0.0"
+
+        updated_quadkeys = set()
+
+        dataset_geoms = polygons(
+            [feat["geometry"]["coordinates"][0] for feat in features]
+        )
+
+        for idx, feature in enumerate(features):
```

Comment:
do we want a progressbar ?

Reply:
Progressbar would be nice for CLI, as long as you can turn it off.  Personally I like the rio-cogeo usage https://github.com/cogeotiff/rio-cogeo/blob/master/rio_cogeo/cogeo.py#L70